src/components/Pill/PillButton/__tests__/PillButton.spec.tsx

```diff
@@ -64,78 +92,115 @@ describe('<PillButton> component', () => {
+            expect(getButtonByName(clearLabel)).not.toHaveClass(
+                'PillButton__button PillButton__button--isOpen'
+            );
+            expect(getButtonByName('This pill is in use clear label')).toHaveClass(
+                'PillButton__pill PillButton__pill--isActive'
+            );
         });
-        it('should not trigger toggle state but onClear only when button is clicked', () => {
-            wrapper.find('.PillButton__button').simulate('click');
+
+        it('should not trigger toggle state but onClear only when button is clicked', async () => {
+            const user = userEvent.setup();
+            await user.click(getButtonByName(clearLabel));
+
             expect(onClearMock).toHaveBeenCalledTimes(1);
             expect(toggleDropdownMock).toHaveBeenCalledTimes(0);
         });
-        it('should not trigger toggle state but onClear only on keyboard interaction with button', () => {
-            wrapper.find('button').simulate('focus').simulate('keyDown', { key: ENTER_KEY });
+
+        it('should not trigger toggle state but onClear only on keyboard interaction with button', async () => {
+            const user = userEvent.setup();
+            const button = getButtonByName(clearLabel);
+            button.focus();
+            await user.keyboard(`[${ENTER_KEY}]`);
+
             expect(onClearMock).toHaveBeenCalledTimes(1);
             expect(toggleDropdownMock).toHaveBeenCalledTimes(0);
         });
     });
+
     describe('in active, open state (content and isOpen prop)', () => {
         beforeEach(() => {
-            wrapper = mount(
+            view = render(
                 <PillButton
                     toggleDropdown={toggleDropdownMock}
                     onClear={onClearMock}
                     name={name}
                     content={content}
                     isOpen
+                    downArrowLabel={downArrowLabel}
+                    upArrowLabel={upArrowLabel}
+                    clearLabel={clearLabel}
                 />
             );
         });
 
         it('should render correctly', () => {
-            expect(toJson(wrapper)).toMatchSnapshot();
-            expect(wrapper.find('.PillButton__pill--isActive')).toHaveLength(1);
-            expect(wrapper.find('.PillButton__pill--isOpen')).toHaveLength(1);
+            expect(view.container).toMatchSnapshot();
+            expect(getButtonByName(upArrowLabel)).toHaveClass(
+                'PillButton__button PillButton__button--isOpen'
+            );
+            expect(getButtonByName('This pill is in use up arrow')).toHaveClass(
+                'PillButton__pill PillButton__pill--isOpen PillButton__pill--isActive'
+            );
         });
+
         it('should have arrow up label', () => {
-            expect(wrapper.find('IoIosArrowDown.PillButton__arrowIcon--isOpen')).toHaveLength(1);
+            expect(screen.getByRole('img')).toHaveClass('PillButton__arrowIcon--isOpen');
```

Comment:
I think the description of the test would be better tested if you check that the button with a down-arrow is present or not

Reply:
yes i've changed it

Reply:
Cannot see the change

Reply:
![image](https://github.com/textkernel/oneui/assets/104063210/3c605ab4-8747-4098-8254-ddd129bdc6fe)

Reply:
It was two cases one of them i've removed

Reply:
but that other test was in a different scenario, no?

Reply:
yes